exec/cnex/global.c

```diff
@@ -731,9 +769,185 @@ void dictionary__keys(TExecutor *exec)
+{
+    BOOL b = top(exec->stack)->boolean; pop(exec->stack);
+    push(exec->stack, cell_fromObject(object_createBooleanObject(b)));
+}
+
+void object__makeBytes(TExecutor *exec)
+{
+    Cell *s = cell_fromCell(top(exec->stack)); pop(exec->stack);
+    push(exec->stack, cell_fromObject(object_fromCell(s)));
+}
+
+void object__makeDictionary(TExecutor *exec)
+{
+    Cell *c = cell_fromCell(top(exec->stack)); pop(exec->stack);
+    push(exec->stack, cell_fromObject(object_fromCell(c)));
+}
+
+void object__makeNull(TExecutor *exec)
+{
+    push(exec->stack, cell_fromObject(object_createObject()));
+}
+
+void object__makeNumber(TExecutor *exec)
 {
     Number v = top(exec->stack)->number; pop(exec->stack);
-    push(exec->stack, cell_fromCString(number_to_string(v)));
+    push(exec->stack, cell_fromObject(object_createNumberObject(v)));
+}
+
+void object__makeString(TExecutor *exec)
+{
+    Cell *s = cell_fromCell(top(exec->stack)); pop(exec->stack);
+    push(exec->stack, cell_fromObject(object_fromCell(s)));
+}
+
+
+void object__isNull(TExecutor *exec)
+{
+    BOOL r = FALSE;
+    Object *o = top(exec->stack)->object;
+
+    if (o == NULL || o->cell == NULL) {
+        r = TRUE;
+    }
+    pop(exec->stack);
+    push(exec->stack, cell_fromBoolean(r));
+}
+
+void object__subscript(struct tagTExecutor *exec)
+{
+    Cell *index = cell_fromCell(top(exec->stack)); pop(exec->stack);
+    Cell *o = cell_fromCell(top(exec->stack)); pop(exec->stack);
+    Cell *r = cell_newCell();
+
+    if (o->object == NULL) {
+        exec->rtl_raise(exec, "DynamicConversionException", "object is null", BID_ZERO);
+        return;
+    }
+    if (o->object->type == oArray) {
+        if (index->object->type != oNumber) {
+            exec->rtl_raise(exec, "DynamicConversionException", "to Number", BID_ZERO);
+            return;
+        }
+        Number i = index->object->cell->number;
+        uint64_t ii = number_to_uint64(i);
+        if (ii >= o->object->cell->array->size) {
+            exec->rtl_raise(exec, "ArrayIndexException", number_to_string(i), BID_ZERO);
+            return;
+        }
+        cell_copyCell(r, &o->object->cell->array->data[ii]);
+    } else if (o->object->type == oDictionary) {
+        if (index->object->type != oString) {
+            exec->rtl_raise(exec, "DynamicConversionException", "to String", BID_ZERO);
+            return;
+        }
+        TString *i = index->object->cell->string;
+        Cell *e = dictionary_findDictionaryEntry(o->object->cell->dictionary, i);
+        if (e == NULL) {
+            char *x = string_asCString(cell_toString(index->object->cell));
+            exec->rtl_raise(exec, "ObjectSubscriptException", x, BID_ZERO);
+            free(x);
+            return;
+        }
+        cell_copyCell(r, e);
+    }
+    cell_freeCell(index);
+    cell_freeCell(o);
+    push(exec->stack, r);
+}
+
+void object__toString(TExecutor *exec)
+{
+    if (top(exec->stack)->object == NULL) {
```

Comment:
This test looks like something that `object_toString` should do, instead of being implemented here.

Reply:
Removed this test, and instead test for NULL in object_toString().